backend/api/v2beta1/job.proto

```diff
@@ -130,10 +156,31 @@ message ListJobsRequest {
   // Ascending by default.
   string sort_by = 3;
 
-  // What resource reference to filter on.
-  // E.g. If listing job for an experiment, the query string would be
-  // resource_reference_key.type=EXPERIMENT&resource_reference_key.id=123
-  ResourceKey resource_reference_key = 4;
+  // Optional input. The namespace the jobs belong to. 
+  string namespace = 4;
```

Comment:
`rpc ListJobs` in v1 covers/supports multiple `resource_types`. So if we want to stay compliant with v1 functionality we need to cover all of them. Please, refer to the document created by @annajung https://docs.google.com/document/d/1ni1hFPgJRMu834oCuihO1R8TxybkMUeSmxbKfZpbuN0/edit#bookmark=id.enhw8uaef8xg
You already have added support for `namespace` and `experiment_id` ( by the new rpc  `ListJobsByExperiment`) .
So I guess we need to add at least support for `version_id` by adding 

>   // Optional input. The version_id the jobs relates to. 
>   string version_id = 5;